peps/pep-9999.rst

```diff
@@ -0,0 +1,616 @@
+behavior in the ecosystem, but new tools have been implementing new
+approaches. Uv and PDM have each added support for some notion of index
+priority.
+
+Index priority is `the default behavior in
+uv <https://docs.astral.sh/uv/pip/compatibility/#packages-that-exist-on-multiple-indexes>`__,
+and supported, but `not default behavior in
+PDM <https://pdm-project.org/latest/usage/config/#respect-the-order-of-the-sources>`__.
+In uv, `users have reported some
+issues <https://github.com/astral-sh/uv/issues/2775>`__ with the
+behavior and asked for keeping pip’s behavior. Uv provides this choice
+as a ``--index-strategy`` flag, with the ``unsafe-best-match`` value
+matching pip’s current behavior, and the ``first-match`` value matching
+this PEP’s proposed behavior. Defining index priority is expected to
+help with `situations where a local package has remote
+dependencies <https://github.com/pypa/pip/issues/11624>`__, and the user
+wishes to prioritize local packages over remote dependencies, while
+still falling back to remote dependencies where needed.
+
+The topic of prioritizing indexes is being proposed as a PEP rather than
+as a change specifically to pip, because predictable index behavior
+across the many tools in our ecosystem is important for consistent,
+predictable behavior between tools.
+
+Specification
+=============
+
+“Version priority”
+------------------
+
+This is pip’s current behavior. All distributions from all configured
+repositories are collated, and the “best” among them is selected. The
+criteria for what is “best” are:
+
+- `Match any specified
+  hashes <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L541>`__
+- `Absence of
+  yanking <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L542>`__
+- `Wheels over
+  sdists <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L504>`__
+- `Maximize
+  version <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L544>`__
+  (including `Local version
+  tags <https://peps.python.org/pep-0440/#local-version-identifiers>`__,
+  `restricted to locally patched private
+  versions <https://discuss.python.org/t/lets-permit-local-version-label-in-version-specifiers/22781/2>`__)
+- `Maximize specificity of platform
+  tag <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L520>`__
+- `Build
+  tags <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L535>`__
+  with the first value being an integer interpreted as ascending values
+  being “better”, and the second being a string with lexical sorting
+
+This behavior is characterized by the installer always getting the
```

Comment:
Perhaps move this to above the bulleted list?  I think this is a good summary of what "version priority" means, so lead with that?